### PR TITLE
feat(query): add security token support for AWS S3

### LINKED ISSUE #7383: Feature: Support to handle session token in COPY INTO command
**Summary**

Description for this feature.
Support session token allow copy into to handle some short lived credentials provisioned by AWS sts assume role. 
ref:
https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html#additional-cloud-provider-parameters

### PR DESCRIPTION
**Pull request by ClSlaid**
Signed-off-by: ClSlaid <cailue@bupt.edu.cn>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

AWS S3 offers an useful feature of temporary credential, that users could authenticate through a rather short-lifed (not longer than a few days) security token strings.

Using short-lifed security tokens avoids the risk of leaking permanent credentials that uncarefully embeded in scripts, worksheets or commit histories. Even if they're leaked, they can not live long to cause as great damage as permanent credentials could done.

## Guide

To use security token in SQLs, just add an `AWS_TOKEN` or `SECURITY_TOKEN` field in `CONNECTION`.

```SQL
COPY INTO books FROM "s3://my_bucket/my_path/books.csv" CONNECTION = (AWS_KEY_ID = "aws_key", AWS_SECRET_KEY = "aws_sec", AWS_TOKEN = "AWS_TOKEN") FILE_FORMAT = (TYPE = "CSV" )
```

Fixes #7383

### CODE REVIEW

**Review by Xuanwo — APPROVED** [review]
Nice PR!

**Comment by Xuanwo** [comment]
Unit tests failed:

```shell
failures:
    configs::test_default_config
    interpreters::interpreter_user_stage::test_user_stage_interpreter
    storages::system::configs_table::test_configs_table
    storages::system::configs_table::test_configs_table_redact
```

**Review by ZhiHanZ — APPROVED** [review]

**Comment by ZhiHanZ** [comment]
Nice work @ClSlaid , could you also update some session token examples on documentation？

File: src/meta/proto-conv/tests/it/user_stage.rs [hunk 1]
```diff
@@ -43,6 +43,55 @@ fn test_user_stage_gcs_latest() -> anyhow::Result<()> {
     Ok(())
 }
 
+#[test]
+fn test_user_stage_s3_v8() -> anyhow::Result<()> {
+    // Encoded data of version 8 of user_stage_s3:
+    // It is generated with common::test_pb_from_to.
+    let user_stage_s3_v8 = vec![
```
Comment (drmingdrmer) [inline src/meta/proto-conv/tests/it/user_stage.rs:50]: The new `VER` value is `9`, I think the test to add should be suffixed with `_v9`?

Comment (Xuanwo) [reply src/meta/proto-conv/tests/it/user_stage.rs:50]: Tracked in https://github.com/datafuselabs/databend/issues/7770

**Pull request merged**